libsolidity/ast/Types.cpp

```diff
@@ -2204,8 +2204,8 @@ string StructType::signatureInExternalFunction(bool _structsByName) const
 		{
 			solAssert(_t, "Parameter should have external type.");
 			auto t = _t->interfaceType(_structsByName);
-			solAssert(t, "");
-			return t->signatureInExternalFunction(_structsByName);
+			solAssert(t.get(), "");
+			return t.get()->signatureInExternalFunction(_structsByName);
```

Comment:
This `.get()` should not be necessary, should it?

Reply:
It is indeed required:
```
Types.cpp:2245:12: Fehler: Basisoperand von »->« hat Nicht-Zeiger-Typ »dev::Result<std::shared_ptr<const dev::solidity::Type> >«
    return t->signatureInExternalFunction(_structsByName);
            ^~
```